sn_networking/src/event.rs

```diff
@@ -831,19 +838,23 @@ impl SwarmDriver {
             } => {
                 event_string = "kad_event::RoutingUpdated";
                 if is_new_peer {
-                    info!("New peer added to routing table: {peer:?}");
+                    let _ = self.connected_peers.saturating_add(1);
+
+                    info!("New peer added to routing table: {peer:?}, now we have #{} connected peers", self.connected_peers);
                     self.log_kbuckets(&peer);
 
                     if self.bootstrap.notify_new_peer() {
                         info!("Performing the first bootstrap");
                         self.initiate_bootstrap();
                     }
-                    self.send_event(NetworkEvent::PeerAdded(peer));
+                    self.send_event(NetworkEvent::PeerAdded(peer, self.connected_peers));
                 }
 
                 if old_peer.is_some() {
+                    let _ = self.connected_peers.saturating_sub(1);
```

Comment:
`saturating_sub` won't change the self value, hence it shall be `self.connected_peers= self.connected_peers.saturating_sub(1);` instead ?